dali/pipeline/data/tensor_vector.cc

```diff
@@ -461,6 +461,8 @@ void TensorVector<Backend>::update_view(int idx) {
     tensors_[idx]->ShareData(std::shared_ptr<void>(ptr, ViewRefDeleter{&views_count_}),
                              volume(tl_->tensor_shape(idx)) * tl_->type().size(), shape,
                              tl_->type());
+  } else if (IsValidType(tl_->type())) {
+    tensors_[idx]->set_type(tl_->type());
```

Comment:
This kept stale type in the views if the pointer or size didn't change.

Reply:
We have Reset above, so it happened only with empty TV that got a type?

Reply:
It happend with something that had non-empty shape and type but 0 volume.